src/test/java/pages/WomenTopsPage.java

```diff
@@ -0,0 +1,68 @@
+package pages;
+
+import io.qameta.allure.Step;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class WomenTopsPage extends AbstractPage {
+
+    private final By colorFilterLocator = By.xpath("//div[@class='filter-options-title' and contains(text(), 'Color')]");
+    private final By selectedColorLocator = By.cssSelector("[aria-label='Color'] [option-label='Blue']");
+    private final By priceFilterLocator = By.xpath("//div[@class='filter-options-title' and contains(text(), 'Price')]");
+    private final By clearFilterLocator = By.cssSelector(".filter-clear");
+    private final By filteredItemsPriceLocator = By.cssSelector(".price-wrapper");
+
+    public WomenTopsPage(WebDriver driver) {
+        super(driver);
+    }
+
+    @Step("Filter items by color")
+    public void filterItemsByColor(String color) {
+        clickElement(colorFilterLocator);
+        clickElement(By.cssSelector("div.filter-options div[option-label='" + color + "']"));
+    }
+
+    @Step("Checking that each item is present in selected color")
+    public void assertEachItemIsInSelectedColor() {
+        List<WebElement> selectedColor = driver.findElements(selectedColorLocator);
+        selectedColor.forEach(element ->
+                assertTrue(element.getAttribute("class").contains("selected"),
```

Comment:
it's better to avoid using 'assertTrue' and 'assertFalse' with logical expressions inside. It returns returns unusable error message on fail. something like 'expected true, actual value false'. I see you provided custom error message but it doesn't give the information what was  in getAttribute("class") and what expected. If junit.jupiter is not flexible enough you could add AssertJ or Hamcrest, or something else

Reply:
Got it, thank you